lib/taskana-core-test/src/test/java/acceptance/taskcomment/query/QueryTaskCommentAccTest.java

```diff
@@ -0,0 +1,457 @@
+      assertThat(listedValues).hasSize(3);
+    }
+
+    @WithAccessId(user = "user-1-1")
+    @Test
+    void should_ReturnListedValues_For_QueryColumnModified() {
+      List<String> listedValues =
+          taskService
+              .createTaskCommentQuery()
+              .listValues(TaskCommentQueryColumnName.MODIFIED, null);
+      assertThat(listedValues).hasSize(3);
+    }
+  }
+
+  @Nested
+  @TestInstance(Lifecycle.PER_CLASS)
+  class OffsetConstraintsTaskComments {
+    @WithAccessId(user = "user-1-1")
+    @Test
+    void should_ConfirmQueryListOffset_When_ProvidingOffsetAndLimit() {
+      List<TaskComment> offsetAndLimitResult = taskService.createTaskCommentQuery().list(1, 2);
+      List<TaskComment> regularResult = taskService.createTaskCommentQuery().list();
+
+      assertThat(offsetAndLimitResult).hasSize(2);
+      assertThat(offsetAndLimitResult.get(0))
+          .isNotEqualTo(regularResult.get(0))
+          .isEqualTo(regularResult.get(1));
+    }
+
+    @WithAccessId(user = "user-1-1")
+    @Test
+    void should_ReturnEmptyList_When_ProvidingWrongConstraints() {
+      List<TaskComment> result = taskService.createTaskCommentQuery().list(0, 1000);
+      assertThat(result).hasSize(3);
+
+      result = taskService.createTaskCommentQuery().list(100, 1000);
+      assertThat(result).isEmpty();
+    }
+  }
+
+  @Nested
+  @TestInstance(Lifecycle.PER_CLASS)
+  class SingleResultTaskComments {
+
+    @WithAccessId(user = "user-1-1")
+    @Test
+    void should_ReturnSingleTaskComment_When_UsingSingleMethod() {
+      TaskComment single =
+          taskService
+              .createTaskCommentQuery()
+              .idIn(comment1.getId())
+              .creatorIn("user-1-1")
+              .single();
+
+      assertThat(single.getId()).isEqualTo(comment1.getId());
+    }
+
+    @WithAccessId(user = "user-1-1")
+    @Test
+    void should_ThrowException_When_SingleMethodRetrievesMoreThanOneEventFromDatabase() {
+      TaskCommentQuery query = taskService.createTaskCommentQuery().creatorIn("user-1-1");
+      assertThatThrownBy(query::single).isInstanceOf(TooManyResultsException.class);
+    }
+  }
+
+  @Nested
+  @TestInstance(Lifecycle.PER_CLASS)
+  class TaskCreatorFullNameWhenAdditionalUserInfoIsTrue implements TaskanaConfigurationModifier {
+
+    @TaskanaInject TaskService taskService;
+    @TaskanaInject UserService userService;
+
+    @Override
+    public TaskanaConfiguration.Builder modify(TaskanaConfiguration.Builder builder) {
+      return builder.addAdditionalUserInfo(true);
+    }
+
+    @WithAccessId(user = "user-1-1")
+    @Test
+    void should_SetTaskCreatorFullNameOfTaskComment_When_PropertyEnabled() throws Exception {
+      List<TaskComment> taskComments =
+          taskService.createTaskCommentQuery().idIn(comment1.getId()).list();
+
+      String creatorFullName = userService.getUser(taskComments.get(0).getCreator()).getFullName();
```

Comment:
Have you checked that the fullName that you get here isn't null? It would be more readable to create the user and the corresponding TaskComment in this class. Or we could check that the fullName isn't null. What do you think?

Reply:
I have changed it so that user will be created in the class and not in the initial setup. I don't think the TaskComments need to be created specificly in each classes, because the same taskcomments can be used by all classes for testing.

I have also added the check if fullname is not null